llvm/lib/Target/SystemZ/SystemZOperands.td

```diff
@@ -262,8 +262,8 @@ def UIMM16 : SDNodeXForm<imm, [{
 
 // Truncate an immediate to a 32-bit signed quantity.
 def SIMM32 : SDNodeXForm<imm, [{
-  return CurDAG->getTargetConstant(int32_t(N->getZExtValue()), SDLoc(N),
-                                   MVT::i64);
+  return CurDAG->getSignedTargetConstant(int32_t(N->getZExtValue()), SDLoc(N),
```

Comment:
```suggestion
  return CurDAG->getSignedTargetConstant(int32_t(N->getSExtValue()), SDLoc(N),
```
Shouldn't make a difference here, but I think this would be more obvious...

Reply:
Good point.  This should be done for all the SIMM variants then, I think.  @redstar would you mind doing this in a follow-on PR?

Reply:
Sure, I'll do that.